atlasdb-impl-shared/src/main/java/com/palantir/atlasdb/transaction/impl/SnapshotTransactionManager.java

```diff
@@ -242,10 +242,18 @@ protected SnapshotTransaction createTransaction(
                 validateLocksOnReads,
                 transactionConfig);
     }
-
     @Override
     public <T, C extends PreCommitCondition, E extends Exception> T runTaskWithConditionReadOnly(
             C condition, ConditionAwareTransactionTask<T, C, E> task) throws E {
+        if (transactionConfig.get().lockImmutableTsOnReadOnlyTransactions()) {
+            return runTaskWithConditionThrowOnConflict(condition, task);
+        } else {
+            return runTaskWithConditionReadOnlyInternal(condition, task);
+        }
```

Comment:
This makes sense.

nit: We don't need `else`; I think I would tend to write this with `? :` notation as well. Any way is fine though!